File: leetcode/easy/1290/1290.js
/**
 * Runtime: 68 ms, faster than 97.66%
 * Memory Usage: 38.6 MB, less than 52.36%
 */

/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function(head) {
  let result = 0;

  while (true) {
    result = result * 2 + head.val;
    if (head.next === null) break;
    head = head.next;
  }

  return result;
};

File: leetcode/easy/1290/1290.rb
# Runtime: 60 ms, faster than 27.73%
# Memory Usage: 209.9 MB, less than 29.41%

# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {Integer}
def get_decimal_value(head)
  result = 0

  while true do
    result = result * 2 + head.val
    head = head.next

    break if head == nil
  end

  result
end
